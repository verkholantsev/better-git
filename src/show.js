// @flow

import checkSupportedOpts from './check-supported-opts';
import mapOpts from './map-opts';
import parseCommits from './parse-commits';

import type { Git } from './git-factory';
import type { Opts } from './map-opts';
import type { Commit } from './parse-commits';

const SUPPORTED_OPTS = [];

export default async function show(git: Git, opts: Opts = {}): Promise<Commit> {
    checkSupportedOpts('show', opts, SUPPORTED_OPTS);

    const args = ['show', ...mapOpts(opts)];
    const out = await git(args);
    const commits = parseCommits(out);

    if (commits.length > 1) {
        throw new Error('Output for `git show` contains more that one commit, it should contain only one');
    } else if (commits.length === 0) {
        throw new Error('Output for `git show` does not contain commits, it should contain exact one');
    }

    return commits[0];
}
