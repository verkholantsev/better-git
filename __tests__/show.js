// @flow

import fs from 'fs';
import path from 'path';
import promisify from 'util.promisify';

import show from '../src/show';

const readFile = promisify(fs.readFile);

describe('git.show()', () => {
    let git;

    beforeEach(() => {
        const fixturePath = path.resolve(__dirname, '__fixtures__', 'git-show');
        git = jest.fn(async () => await readFile(fixturePath, { encoding: 'utf8' }));
    });

    it('should produce correct output', async () => {
        expect(await show(git)).toMatchSnapshot();
    });

    it('should call git with correct args', async () => {
        await show(git);
        expect(git).toBeCalledWith(['show']);
    });
});
