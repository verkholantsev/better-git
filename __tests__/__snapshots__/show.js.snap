// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`git.show() should produce correct output 1`] = `
Object {
  "author": "dependabot[bot] <support@dependabot.com>",
  "changedFiles": Array [
    Object {
      "destination": "package.json",
      "source": "package.json",
    },
    Object {
      "destination": "yarn.lock",
      "source": "yarn.lock",
    },
  ],
  "commit": "88e99b0809d9b73021385988c316173c9f10e8f0",
  "date": "Thu Mar 15 19:11:29 2018 +0000",
  "diff": "diff --git a/package.json b/package.json
index e59e7e4..391800c 100644
--- a/package.json
+++ b/package.json
@@ -47,7 +47,7 @@
-        \\"rollup\\": \\"^0.56.2\\",
+        \\"rollup\\": \\"^0.57.0\\",
diff --git a/yarn.lock b/yarn.lock
index c228ec7..6e3fdfe 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -71,6 +71,16 @@
 
+\\"@types/acorn@^4.0.3\\":
+  version \\"4.0.3\\"
+  resolved \\"https://registry.yarnpkg.com/@types/acorn/-/acorn-4.0.3.tgz#d1f3e738dde52536f9aad3d3380d14e448820afd\\"
+  dependencies:
+    \\"@types/estree\\" \\"*\\"
+
+\\"@types/estree@*\\", \\"@types/estree@0.0.38\\":
+  version \\"0.0.38\\"
+  resolved \\"https://registry.yarnpkg.com/@types/estree/-/estree-0.0.38.tgz#c1be40aa933723c608820a99a373a16d215a1ca2\\"
+
 abab@^1.0.4:
   version \\"1.0.4\\"
   resolved \\"https://registry.yarnpkg.com/abab/-/abab-1.0.4.tgz#5faad9c2c07f60dd76770f71cf025b62a63cfd4e\\"
@@ -79,6 +89,12 @@ abbrev@1:
   version \\"1.1.1\\"
   resolved \\"https://registry.yarnpkg.com/abbrev/-/abbrev-1.1.1.tgz#f8f2c887ad10bf67f634f005b6987fed3179aac8\\"
 
+acorn-dynamic-import@^3.0.0:
+  version \\"3.0.0\\"
+  resolved \\"https://registry.yarnpkg.com/acorn-dynamic-import/-/acorn-dynamic-import-3.0.0.tgz#901ceee4c7faaef7e07ad2a47e890675da50a278\\"
+  dependencies:
+    acorn \\"^5.0.0\\"
+
 acorn-globals@^4.1.0:
   version \\"4.1.0\\"
   resolved \\"https://registry.yarnpkg.com/acorn-globals/-/acorn-globals-4.1.0.tgz#ab716025dbe17c54d3ef81d32ece2b2d99fe2538\\"
@@ -99,6 +115,10 @@ acorn@^5.0.0, acorn@^5.3.0, acorn@^5.4.0:
   version \\"5.4.1\\"
   resolved \\"https://registry.yarnpkg.com/acorn/-/acorn-5.4.1.tgz#fdc58d9d17f4a4e98d102ded826a9b9759125102\\"
 
+acorn@^5.5.3:
+  version \\"5.5.3\\"
+  resolved \\"https://registry.yarnpkg.com/acorn/-/acorn-5.5.3.tgz#f473dd47e0277a08e28e9bec5aeeb04751f0b8c9\\"
+
 ajv-keywords@^2.1.0:
   version \\"2.1.1\\"
   resolved \\"https://registry.yarnpkg.com/ajv-keywords/-/ajv-keywords-2.1.1.tgz#617997fc5f60576894c435f940d819e135b80762\\"
@@ -1222,6 +1242,12 @@ date-fns@^1.27.2:
   version \\"1.29.0\\"
   resolved \\"https://registry.yarnpkg.com/date-fns/-/date-fns-1.29.0.tgz#12e609cdcb935127311d04d33334e2960a2a54e6\\"
 
+date-time@^2.1.0:
+  version \\"2.1.0\\"
+  resolved \\"https://registry.yarnpkg.com/date-time/-/date-time-2.1.0.tgz#0286d1b4c769633b3ca13e1e62558d2dbdc2eba2\\"
+  dependencies:
+    time-zone \\"^1.0.0\\"
+
 debug@^2.2.0, debug@^2.6.8:
   version \\"2.6.9\\"
   resolved \\"https://registry.yarnpkg.com/debug/-/debug-2.6.9.tgz#5d128515df134ff327e90a4c93f4e077a536341f\\"
@@ -2123,6 +2149,10 @@ invert-kv@^1.0.0:
   version \\"1.0.0\\"
   resolved \\"https://registry.yarnpkg.com/invert-kv/-/invert-kv-1.0.0.tgz#104a8e4aaca6d3d8cd157a8ef8bfab2d7a3ffdb6\\"
 
+irregular-plurals@^1.0.0:
+  version \\"1.4.0\\"
+  resolved \\"https://registry.yarnpkg.com/irregular-plurals/-/irregular-plurals-1.4.0.tgz#2ca9b033651111855412f16be5d77c62a458a766\\"
+
 is-arrayish@^0.2.1:
   version \\"0.2.1\\"
   resolved \\"https://registry.yarnpkg.com/is-arrayish/-/is-arrayish-0.2.1.tgz#77c99840527aa8ecb1a8ba697b80645a7a926a9d\\"
@@ -2262,6 +2292,12 @@ is-redirect@^1.0.0:
   version \\"1.0.0\\"
   resolved \\"https://registry.yarnpkg.com/is-redirect/-/is-redirect-1.0.0.tgz#1d03dded53bd8db0f30c26e4f95d36fc7c87dc24\\"
 
+is-reference@^1.1.0:
+  version \\"1.1.0\\"
+  resolved \\"https://registry.yarnpkg.com/is-reference/-/is-reference-1.1.0.tgz#50e6ef3f64c361e2c53c0416cdc9420037f2685b\\"
+  dependencies:
+    \\"@types/estree\\" \\"0.0.38\\"
+
 is-regex@^1.0.4:
   version \\"1.0.4\\"
   resolved \\"https://registry.yarnpkg.com/is-regex/-/is-regex-1.0.4.tgz#5517489b547091b0930e095654ced25ee97e9491\\"
@@ -2866,6 +2902,10 @@ load-json-file@^4.0.0:
 
+locate-character@^2.0.5:
+  version \\"2.0.5\\"
+  resolved \\"https://registry.yarnpkg.com/locate-character/-/locate-character-2.0.5.tgz#f2d2614d49820ecb3c92d80d193b8db755f74c0f\\"
+
 locate-path@^2.0.0:
   version \\"2.0.0\\"
   resolved \\"https://registry.yarnpkg.com/locate-path/-/locate-path-2.0.0.tgz#2b568b265eec944c6d9c0de9c3dbbbca0354cd8e\\"
@@ -3343,6 +3383,10 @@ parse-json@^4.0.0:
 
+parse-ms@^1.0.0:
+  version \\"1.0.1\\"
+  resolved \\"https://registry.yarnpkg.com/parse-ms/-/parse-ms-1.0.1.tgz#56346d4749d78f23430ca0c713850aef91aa361d\\"
+
 parse5@4.0.0:
   version \\"4.0.0\\"
   resolved \\"https://registry.yarnpkg.com/parse5/-/parse5-4.0.0.tgz#6d78656e3da8d78b4ec0b906f7c08ef1dfe3f608\\"
@@ -3423,6 +3467,12 @@ platform@^1.3.3:
   version \\"1.3.5\\"
   resolved \\"https://registry.yarnpkg.com/platform/-/platform-1.3.5.tgz#fb6958c696e07e2918d2eeda0f0bc9448d733444\\"
 
+plur@^2.1.2:
+  version \\"2.1.2\\"
+  resolved \\"https://registry.yarnpkg.com/plur/-/plur-2.1.2.tgz#7482452c1a0f508e3e344eaec312c91c29dc655a\\"
+  dependencies:
+    irregular-plurals \\"^1.0.0\\"
+
 pluralize@^7.0.0:
   version \\"7.0.0\\"
   resolved \\"https://registry.yarnpkg.com/pluralize/-/pluralize-7.0.0.tgz#298b89df8b93b0221dbf421ad2b1b1ea23fc6777\\"
@@ -3454,6 +3504,13 @@ pretty-format@^22.4.0:
 
+pretty-ms@^3.1.0:
+  version \\"3.1.0\\"
+  resolved \\"https://registry.yarnpkg.com/pretty-ms/-/pretty-ms-3.1.0.tgz#e9cac9c76bf6ee52fe942dd9c6c4213153b12881\\"
+  dependencies:
+    parse-ms \\"^1.0.0\\"
+    plur \\"^2.1.2\\"
+
 private@^0.1.6, private@^0.1.7:
   version \\"0.1.8\\"
   resolved \\"https://registry.yarnpkg.com/private/-/private-0.1.8.tgz#2381edb3689f7a53d653190060fcf822d2f368ff\\"
@@ -3742,6 +3799,10 @@ require-main-filename@^1.0.1:
   version \\"1.0.1\\"
   resolved \\"https://registry.yarnpkg.com/require-main-filename/-/require-main-filename-1.0.1.tgz#97f717b69d48784f5f526a6c5aa8ffdda055a4d1\\"
 
+require-relative@^0.8.7:
+  version \\"0.8.7\\"
+  resolved \\"https://registry.yarnpkg.com/require-relative/-/require-relative-0.8.7.tgz#7999539fc9e047a37928fa196f8e1563dabd36de\\"
+
 require-uncached@^1.0.3:
   version \\"1.0.3\\"
   resolved \\"https://registry.yarnpkg.com/require-uncached/-/require-uncached-1.0.3.tgz#4e0d56d6c9662fd31e43011c4b95aa49955421d3\\"
@@ -3842,9 +3903,21 @@ rollup-pluginutils@^2.0.1:
 
-rollup@^0.56.2:
-  version \\"0.56.5\\"
-  resolved \\"https://registry.yarnpkg.com/rollup/-/rollup-0.56.5.tgz#40fe3cf0cd1659d469baad11f4d5b6336c14ce84\\"
+rollup@^0.57.0:
+  version \\"0.57.0\\"
+  resolved \\"https://registry.yarnpkg.com/rollup/-/rollup-0.57.0.tgz#0a57884477175592efe0c8bfc408ea12aa4c0f21\\"
+  dependencies:
+    \\"@types/acorn\\" \\"^4.0.3\\"
+    acorn \\"^5.5.3\\"
+    acorn-dynamic-import \\"^3.0.0\\"
+    date-time \\"^2.1.0\\"
+    is-reference \\"^1.1.0\\"
+    locate-character \\"^2.0.5\\"
+    pretty-ms \\"^3.1.0\\"
+    require-relative \\"^0.8.7\\"
+    rollup-pluginutils \\"^2.0.1\\"
+    signal-exit \\"^3.0.2\\"
+    sourcemap-codec \\"^1.4.1\\"
 
 run-async@^2.2.0:
   version \\"2.3.0\\"
@@ -3986,6 +4059,10 @@ source-map@^0.6.0:
   version \\"0.6.1\\"
   resolved \\"https://registry.yarnpkg.com/source-map/-/source-map-0.6.1.tgz#74722af32e9614e9c287a8d0bbde48b5e2f1a263\\"
 
+sourcemap-codec@^1.4.1:
+  version \\"1.4.1\\"
+  resolved \\"https://registry.yarnpkg.com/sourcemap-codec/-/sourcemap-codec-1.4.1.tgz#c8fd92d91889e902a07aee392bdd2c5863958ba2\\"
+
 spdx-correct@~1.0.0:
   version \\"1.0.2\\"
   resolved \\"https://registry.yarnpkg.com/spdx-correct/-/spdx-correct-1.0.2.tgz#4b3073d933ff51f3912f03ac5519498a4150db40\\"
@@ -4228,6 +4305,10 @@ through@2, through@^2.3.6, through@^2.3.8, through@~2.3.4:
   version \\"2.3.8\\"
   resolved \\"https://registry.yarnpkg.com/through/-/through-2.3.8.tgz#0dd4c9ffaabc357960b1b724115d7e0e86a2e1f5\\"
 
+time-zone@^1.0.0:
+  version \\"1.0.0\\"
+  resolved \\"https://registry.yarnpkg.com/time-zone/-/time-zone-1.0.0.tgz#99c5bf55958966af6d06d83bdf3800dc82faec5d\\"
+
 timed-out@^4.0.0:
   version \\"4.0.1\\"
   resolved \\"https://registry.yarnpkg.com/timed-out/-/timed-out-4.0.1.tgz#f32eacac5a175bea25d7fab565ab3ed8741ef56f\\"",
  "message": "Bump rollup from 0.56.5 to 0.57.0

Bumps [rollup](https://github.com/rollup/rollup) from 0.56.5 to 0.57.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v0.56.5...v0.57.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>
     \\"lodash\\": \\"^4.17.5\\",
     \\"np\\": \\"^2.20.1\\",
     \\"prettier\\": \\"^1.10.2\\",
     \\"rollup-plugin-babel\\": \\"^3.0.3\\",
     \\"rollup-plugin-commonjs\\": \\"^9.0.0\\",
     \\"rollup-plugin-node-resolve\\": \\"^3.0.3\\"
 lodash \\"^4.2.0\\"
 to-fast-properties \\"^2.0.0\\"
 pify \\"^3.0.0\\"
 strip-bom \\"^3.0.0\\"
 error-ex \\"^1.3.1\\"
 json-parse-better-errors \\"^1.0.1\\"
 ansi-regex \\"^3.0.0\\"
 ansi-styles \\"^3.2.0\\"
 estree-walker \\"^0.3.0\\"
 micromatch \\"^2.3.11\\"",
}
`;
